controller/node_controller.go

```diff
@@ -1201,17 +1208,17 @@ func (nc *NodeController) cleanUpBackingImagesInDisks(node *longhorn.Node) error
 	return nil
 }
 
-func BackingImageDiskFileCleanup(node *longhorn.Node, bi *longhorn.BackingImage, bids *longhorn.BackingImageDataSource, waitInterval time.Duration, haRequirement int) {
-	if bi.Spec.Disks == nil || bi.Status.DiskLastRefAtMap == nil {
+func BackingImageDiskFileCleanup(node *longhorn.Node, bi *longhorn.BackingImage, bids *longhorn.BackingImageDataSource, waitInterval time.Duration, minNumberOfCopies int) {
+	if bi.Spec.DiskFileSpecMap == nil || bi.Status.DiskLastRefAtMap == nil || !bids.Spec.FileTransferred {
```

Comment:
bids is to prepare the first backing file
when it is ready it will be transferred to backing image manager

backing image can only be used when it is transferred
so we should only check if we need to cleanup the backing image file after it is transferred

original code did the same thing, I just moved it to the very beginning.
```
		isFirstFile := bids != nil && !bids.Spec.FileTransferred && diskUUID == bids.Spec.DiskUUID
		if isFirstFile {
			continue
		}
```